protocol/directory.go

```diff
@@ -70,13 +74,13 @@ func (d *ConiksDirectory) Register(req *RegistrationRequest) (
 		return nil, ErrorDirectory
 	}
 	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
-		return NewRegistrationProof(ap, d.LatestSTR(), nil), ErrorNameExisted
+		return NewRegistrationProof(ap, d.LatestSTR(), nil, ErrorNameExisted)
```

Comment:
Maybe using the prefix `Error` here is not right choice but all our predefined `ErrorXxx`s are used for logging and returning to the client only, that is `ErrorNameExisted` and `ErrorDirectory` are treated as the same.